lib/iris/experimental/regrid.py

```diff
@@ -1105,9 +1252,25 @@ def __call__(self, src):
                              'source grid as this regridder.')
 
         # Call the regridder function.
-        res = regrid_weighted_curvilinear_to_rectilinear(src, self.weights,
-                                                         self._target_cube)
-        return res
+        # This includes repeating over any non-XY dimensions, because the
+        # underlying routine does not support this.
+        # FOR NOW: we will use cube.slices and merge to achieve this,
+        # though that is not a terribly efficient method ...
```

Comment:
Dead right.  
In fact, I think it could be a useful 'general Iris tidy' to rationalise how this is done in a bunch of different places.  Using a common code structure, helper routines or whatever.